### DISCUSSION TITLE
Support Elasticsearch API key authentication

### DISCUSSION THREAD

**Discussion by cfeltz34** [fluent/fluent-bit#9491] [Ideas]
There's no Elasticsearch API key option in here: https://docs.fluentbit.io/manual/pipeline/outputs/elasticsearch

Is there a way to use an api key by setting the header or by setting the api key.

example:
Header Authorization ${ELASTICSEARCH_APIKEY}
or
api_key: ${ELASTICSEARCH_APIKEY}

See also : https://github.com/fluent/fluent-bit/issues/8357

**Comment by patrick-stephens** [comment]
Presumably not Elastic Cloud? https://docs.fluentbit.io/manual/pipeline/outputs/elasticsearch#fluent-bit--elastic-cloud
Just checking - I don't use it but there is a way to pass Elastic Cloud creds if that's what you were after.
